### PR TITLE
48017_trainee_home

### PR DESCRIPTION
**Pull request by zenith19**

### CODE REVIEW

File: app/assets/stylesheets/courses.scss [hunk 1]
```diff
@@ -0,0 +1,3 @@
+// Place all the styles related to the courses controller here.
+// They will automatically be included in application.css.
+// You can use Sass (SCSS) here: http://sass-lang.com/
```
Comment (rofiuddin) [inline app/assets/stylesheets/courses.scss]: remove

File: app/assets/javascripts/courses.coffee [hunk 2]
```diff
@@ -0,0 +1,3 @@
+# Place all the behaviors and hooks related to the matching controller here.
+# All this logic will automatically be available in application.js.
+# You can use CoffeeScript in this file: http://coffeescript.org/
```
Comment (rofiuddin) [inline app/assets/javascripts/courses.coffee]: remove

File: app/models/user.rb [hunk 3]
```diff
@@ -6,18 +6,20 @@ class User < ActiveRecord::Base
     supervisor: '1'
```
Comment (abdullahallmehedi) [inline app/models/user.rb]: ""

File: config/locales/en.yml [hunk 4]
```diff
@@ -1,15 +1,21 @@
...
-  finished: "Finished"
-  tasks: "Tasks:"
   button:
     hide: Hide
     show: Show
+  confirm_message: You sure?
   contact: Contact
+  courses:
+    show:
+      course_name: 'Course Name: '
```
Comment (abdullahallmehedi) [inline config/locales/en.yml]: ""

File: config/locales/en.yml [hunk 5]
```diff
@@ -1,15 +1,21 @@
...
+      no_of_supervisors: ' supervisor(s)'
```
Comment (abdullahallmehedi) [inline config/locales/en.yml]: ""

File: config/routes.rb [hunk 6]
```diff
@@ -9,7 +9,8 @@
 
   resources :subjects
   resources :users, only: [:index, :show]
-  
+  resources :courses, only: [:show]
+
   namespace :supervisor do
     root 'courses#index'
```
Comment (abdullahallmehedi) [inline config/routes.rb]: ""

File: app/models/user.rb [hunk 7]
```diff
@@ -2,22 +2,24 @@ class User < ActiveRecord::Base
...
+  has_many :subjects, through: :courses_subjects
   has_many :tasks, through: :users_tasks
-
+  
   scope :supervisors, -> { where(supervisor: true) }
   scope :trainees, -> { where(supervisor: false) }
 
   def full_name
-    first_name+ " " + last_name
+    first_name + " " + last_name
```
Comment (dieunb) [inline app/models/user.rb]: should try `[first_name, last_name].join(" ")`

File: app/views/courses/show.html.erb [hunk 8]
```diff
@@ -0,0 +1,35 @@
...
+  <h4><%= @course.users.supervisors.count %><%= t ".no_of_supervisors" %> </h4>
+<div id="show_hide_supervisor">
+<table class="table table-striped">
+  <thead>
+    <th><%= t ".supervisor_name" %></th>
+    <th><%= t ".supervisor_email" %></th>
+  </thead>
+  <tbody>
+    <% @course.supervisors.each do |supervisor| %>
+    <tr>
```
Comment (dieunb) [inline app/views/courses/show.html.erb]: indent

File: app/views/courses/show.html.erb [hunk 9]
```diff
@@ -0,0 +1,35 @@
...
+      <td><%= supervisor.first_name << " " << supervisor.last_name %></td>
+      <td><%= supervisor.email %></td>
+    </tr>
```
Comment (dieunb) [inline app/views/courses/show.html.erb]: indent

File: app/views/courses/show.html.erb [hunk 10]
```diff
@@ -0,0 +1,35 @@
...
+<div id="show_hide_supervisor">
+<table class="table table-striped">
+  <thead>
+    <th><%= t ".supervisor_name" %></th>
+    <th><%= t ".supervisor_email" %></th>
+  </thead>
+  <tbody>
+    <% @course.supervisors.each do |supervisor| %>
+    <tr>
+      <td><%= supervisor.first_name << " " << supervisor.last_name %></td>
```
Comment (dieunb) [inline app/views/courses/show.html.erb]: call fullname

**Pull request merged**